### PR TITLE
BlockSetTypeRegistry and WoodTypeRegistry

### PR DESCRIPTION
**Pull request by Shnupbups**
Adds `BlockSetTypeRegistry` for creating `BlockSetType`s using Identifiers. Also adds `WoodTypeRegistry` to replace `SignTypeRegistry`, matching the new more accurate Yarn name.

Also updates Yarn build.

### CODE REVIEW

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	private BlockSetTypeRegistry() {
+	}
+
+	/**
+	 * Creates and registers a {@link BlockSetType}.
+	 *
+	 * @param id the id of this {@link BlockSetType}
+	 * @return a new {@link BlockSetType}.
+	 */
+	public static BlockSetType registerBlockSetType(Identifier id) {
```
Comment (apple502j) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java]: This should be renamed; the block set created is specifically for overworld woods.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public final class BlockSetTypeRegistry {
+	private BlockSetTypeRegistry() {
+	}
+
+	/**
+	 * Creates and registers a {@link BlockSetType}.
+	 *
+	 * @param id the id of this {@link BlockSetType}
+	 * @return a new {@link BlockSetType}.
```
Comment (apple502j) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java]: I don't think `@return` should end with a dot? (same for others)

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+
+/**
+ * This class allows registering {@link WoodType}s.
+ *
+ * <p>A {@link WoodType} is used to tell the game what textures signs should use, as well as sounds for both signs and fence gates.
+ *
+ * <p>Regular sign textures are stored as {@code [namespace]/textures/entity/signs/[path].png}.
+ * <br>Hanging sign textures are stored as {@code [namespace]/textures/entity/signs/hanging/[path].png}.
```
Comment (apple502j) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java]: Stored at, not as

**Review by apple502j — CHANGES_REQUESTED** [review]

**Review by Technici4n — APPROVED** [review]

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	private BlockSetTypeRegistry() {
+	}
+
+	/**
+	 * Creates and registers a {@link BlockSetType} with the regular wood sounds.
+	 *
+	 * @param id the id of this {@link BlockSetType}
+	 * @return a new {@link BlockSetType}
+	 */
+	public static BlockSetType registerWoodBlockSetType(Identifier id) {
```
Comment (Technici4n) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java]: Actually, can't we just call this one `registerWood`

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	 * @param doorOpen the {@link SoundEvent} for when this block set's door opens
+	 * @param trapdoorClose the {@link SoundEvent} for when this block set's trapdoor closes
+	 * @param trapdoorOpen the {@link SoundEvent} for when this block set's trapdoor opens
+	 * @param pressurePlateClickOff the {@link SoundEvent} for when this block set's pressure plate is unpressed
+	 * @param pressurePlateClickOn the {@link SoundEvent} for when this block set's pressure plate is pressed
+	 * @param buttonClickOff the {@link SoundEvent} for when this block set's button is unpressed
+	 * @param buttonClickOn the {@link SoundEvent} for when this block set's button is pressed
+	 * @return a new {@link BlockSetType}
+	 */
+	public static BlockSetType registerBlockSetType(Identifier id, BlockSoundGroup soundType, SoundEvent doorClose, SoundEvent doorOpen, SoundEvent trapdoorClose, SoundEvent trapdoorOpen, SoundEvent pressurePlateClickOff, SoundEvent pressurePlateClickOn, SoundEvent buttonClickOff, SoundEvent buttonClickOn) {
```
Comment (Technici4n) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java]: And this one `register`?

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	/**
+	 * Creates and registers a {@link WoodType}.
+	 *
+	 * @param id the id of this {@link WoodType}
+	 * @param setType the {@link BlockSetType} for this wood type
+	 * @return a new {@link WoodType}
+	 */
+	public static WoodType registerWoodType(Identifier id, BlockSetType setType) {
```
Comment (Technici4n) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java]: Just `register`?

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	 *
+	 * @param id the id of this {@link WoodType}
+	 * @param setType the {@link BlockSetType} for this wood type
+	 * @param soundType the {@link BlockSoundGroup} for this wood type
+	 * @param hangingSignSoundType the {@link BlockSoundGroup} for this wood type's hanging sign
+	 * @param fenceGateClose the {@link SoundEvent} for when this wood type's fence gate closes
+	 * @param fenceGateOpen the {@link SoundEvent} for when this wood type's fence gate opens
+	 * @return a new {@link WoodType}
+	 */
+	public static WoodType registerWoodType(Identifier id, BlockSetType setType, BlockSoundGroup soundType, BlockSoundGroup hangingSignSoundType, SoundEvent fenceGateClose, SoundEvent fenceGateOpen) {
```
Comment (Technici4n) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/WoodTypeRegistry.java]: Just `register`?

**Review by Technici4n — CHANGES_REQUESTED** [review]
Just noticed that the method names are redundant... :smile:

**Review by Technici4n — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**